materia/views.py

```diff
@@ -1187,117 +1191,92 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class MateriaLegislativaPesquisaView(FormView):
-    template_name = 'materia/pesquisa_materia.html'
-
-    def get_success_url(self):
-        return reverse('pesquisar_materia')
-
-    def get(self, request, *args, **kwargs):
-        form = MateriaLegislativaPesquisaForm()
-        return self.render_to_response({'form': form})
-
-    def post(self, request, *args, **kwargs):
-        kwargs = {}
-        form = MateriaLegislativaPesquisaForm(request.POST)
-
-        if form.data['tipo']:
-            kwargs['tipo'] = form.data['tipo']
-
-        if form.data['numero']:
-            kwargs['numero'] = form.data['numero']
-
-        if form.data['ano']:
-            kwargs['ano'] = form.data['ano']
-
-        if form.data['numero_protocolo']:
-            kwargs['numero_protocolo'] = form.data['numero_protocolo']
-
-        if (form.data['apresentacao_inicial'] and
-                form.data['apresentacao_final']):
-            kwargs['apresentacao_inicial'] = form.data['apresentacao_inicial']
-            kwargs['apresentacao_final'] = form.data['apresentacao_final']
-
-        if (form.data['publicacao_inicial'] and
-                form.data['publicacao_final']):
-            kwargs['publicacao_inicial'] = form.data['publicacao_inicial']
-            kwargs['publicacao_final'] = form.data['publicacao_final']
-
-        if form.data['local_origem_externa']:
-            kwargs['local_origem_externa'] = form.data['local_origem_externa']
-
-        if form.data['autor']:
-            kwargs['autor'] = form.data['autor']
+class MateriaLegislativaPesquisaView(FilterView):
+    model = MateriaLegislativa
+    filterset_class = MateriaLegislativaPesquisaFields
+    paginate_by = 10
 
-        if form.data['localizacao']:
-            kwargs['localizacao'] = form.data['localizacao']
+    def get_context_data(self, **kwargs):
+        context = super(MateriaLegislativaPesquisaView,
+                        self).get_context_data(**kwargs)
 
-        if form.data['em_tramitacao']:
-            kwargs['em_tramitacao'] = form.data['em_tramitacao']
+        paginator = context['paginator']
+        page_obj = context['page_obj']
 
-        if form.data['situacao']:
-            kwargs['situacao'] = form.data['situacao']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        return context
 
-        request.session['kwargs'] = kwargs
-        return redirect('pesquisar_materia_list')
+    def get(self, request, *args, **kwargs):
+        filterset_class = self.get_filterset_class()
+        self.filterset = self.get_filterset(filterset_class)
+
+        status_tramitacao = self.filterset.data.get('tramitacao__status')
+        unidade_destino = self.filterset.data.get(
+            'tramitacao__unidade_tramitacao_destino')
+        if (status_tramitacao and status_tramitacao != '' and
```

Comment:
Basta usar ``if status_tramitacao and unidade_destino:`` (não precisa do ``!= ''`` porque se for vazio ou None vai ser false.